storage/client/queue/src/main/java/com/azure/storage/queue/QueueClientBuilder.java

```diff
@@ -0,0 +1,339 @@
+        policies.add(new HttpLoggingPolicy(logLevel));
+
+        HttpPipeline pipeline = HttpPipeline.builder()
+            .policies(policies.toArray(new HttpPipelinePolicy[0]))
+            .httpClient(httpClient)
+            .build();
+
+        return new QueueAsyncClient(endpoint, pipeline, queueName);
+    }
+
+    /**
+     * Sets the endpoint for the Azure Storage Queue instance that the client will interact with.
+     *
+     * <p>The first path segment, if the endpoint contains path segments, will be assumed to be the name of the queue
+     * that the client will interact with.</p>
+     *
+     * <p>Query parameters of the endpoint will be parsed using {@link SASTokenCredential#fromQuery(String) fromQuery} in an
+     * attempt to generate a {@link SASTokenCredential} to authenticate requests sent to the service.</p>
+     *
+     * @param endpoint The URL of the Azure Storage Queue instance to send service requests to and receive responses from.
+     * @return the updated QueueClientBuilder object
+     * @throws IllegalArgumentException If {@code endpoint} isn't a proper URL
+     */
+    public QueueClientBuilder endpoint(String endpoint) {
+        Objects.requireNonNull(endpoint);
+        try {
+            URL fullURL = new URL(endpoint);
+            this.endpoint = new URL(fullURL.getProtocol() + "://" + fullURL.getHost());
+
+            // Attempt to get the queue name from the URL passed
+            String[] pathSegments = fullURL.getPath().split("/", 2);
+            if (pathSegments.length == 2 && !ImplUtils.isNullOrEmpty(pathSegments[1])) {
+                this.queueName = pathSegments[1];
+            }
+
+            // Attempt to get the SAS token from the URL passed
+            SASTokenCredential credential = SASTokenCredential.fromQuery(fullURL.getQuery());
+            if (credential != null) {
+                this.sasTokenCredential = credential;
+            }
+        } catch (MalformedURLException ex) {
+            throw new IllegalArgumentException("The Azure Storage Queue endpoint url is malformed.");
+        }
+
+        return this;
+    }
+
+    /**
+     * Sets the name of the queue that the client will interact with.
+     *
+     * @param queueName Name of the queue
+     * @return the updated QueueClientBuilder object
+     * @throws NullPointerException If {@code queueName} is {@code null}.
+     */
+    public QueueClientBuilder queueName(String queueName) {
+        this.queueName = Objects.requireNonNull(queueName);
+        return this;
+    }
+
+    /**
+     * Sets the {@link SASTokenCredential} used to authenticate requests sent to the Queue service.
+     *
+     * @param credential SAS token credential generated from the Storage account that authorizes requests
+     * @return the updated QueueClientBuilder object
+     * @throws NullPointerException If {@code credential} is {@code null}.
+     */
+    public QueueClientBuilder credential(SASTokenCredential credential) {
```

Comment:
Should add similar overloads for credential that Blob has, SharedKeyCredential, TokenCredential, and AnonymousCredential

Reply:
I will add in both queue and queue service

Reply:
Do we add DefailtAzureCredential?

Reply:
Done.